### PR TITLE
Set xterm 16 colors for themes when they are available

### PR DESCRIPTION
**Pull request by jimhester**

### CODE REVIEW

File: src/gwt/tools/compile-themes.R [hunk 1]
```diff
@@ -189,6 +211,7 @@ themeFiles <- list.files(
 ## Process the theme files -- we strip out the name preceeding the theme,
 ## and then add some custom rules.
 for (themeFile in themeFiles) {
+  str(themeFile)
```
Comment (jimhester) [inline src/gwt/tools/compile-themes.R]: This is just here for debugging, I will remove it once https://github.com/rstudio/rstudio/issues/3907 is sorted.

**Comment by jmcphers** [comment]
Note that before we can take this change, we'll need to fix https://github.com/rstudio/rstudio/issues/3907 and regenerate the internal themes.

**Comment by gtritchie** [comment]
Would also be nice to fix #4009

**Comment by jmcphers** [comment]
@jimhester Are you able to regen the themes now that #3907 is fixed, or do you need someone from the dev team to take over?

**Comment by jimhester** [comment]
Hmm the fix in #3907 doesn't seem to fix the issue for me, the error is the same as it was previously.

**Comment by jmcphers** [comment]
You're right, it doesn't look like #3907 is fixed. I've reopened it.

**Comment by jmcphers** [comment]
@jimhester I think I've got #3907 sorted now (and was able to successfully compile themes in your branch). Let us know if it's still an issue!

**Comment by jimhester** [comment]
So I spent some time looking at this again, now that #3907 is fixed the code now runs without failing. However unfortunately the newer version of ace now uses [group css selectors](http://www.websiteoptimization.com/speed/tweak/groupsel/) which the [parseCss](https://github.com/rstudio/rstudio/blob/b6d30146e9f3e798dd73c3325d096ea2b36d9d07/src/cpp/session/resources/themes/compile-themes.R#L15-L162) regexes do not handle. e.g. I see a few options on how to get past this.

1. Add additional logic to `parseCss` to handle this, will be a bit tricky to do because the function operates line-wise and the logic is already getting pretty unwieldy (IMHO).
2. Use something else to parse the css into a AST, which can then be queried. I don't think there are any R packages to do this, but could probably use the V8 package plus the [css](https://www.npmjs.com/package/css) npm module.
3. Edit the css files so they work with the existing code, either manually or with a pre-processing script.

For 1. I am not sure if I would be the best person to work on it, maybe it would be a something for @MariaSemple to handle, since she wrote the original code.
2. would be the most robust to future changes in the CSS, but would also be the most heavy solution, and I am not sure I have enough free bandwidth to take on for this.
3. I could probably get something together that would get us past this issue, but I am not sure that is the route you want to take as it would complicate the task of recompiling the scripts.

**Comment by jmcphers** [comment]
Since there's no clear path forward with this changeset, I'm closing this PR. We will use #1959 to track the work needed to get his done.

**Pull request closed without merging**